File: src/main/java/nl/rabobank/kotlinmovement/recipes/model/RecipeRequest.java
package nl.rabobank.kotlinmovement.recipes.model;


import lombok.Data;

import javax.validation.Valid;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.util.Set;

@Data
public class RecipeRequest {
    @NotBlank( message = "recipeName")
    private final String recipeName;
    @NotEmpty(message = "ingredients")
    @Valid
    private final Set<IngredientRequest> ingredients;
}

File: java-to-kotlin-complete/src/main/kotlin/nl/rabobank/kotlinmovement/recipes/model/RecipeRequest.kt
package nl.rabobank.kotlinmovement.recipes.model

import javax.validation.Valid
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotEmpty

data class RecipeRequest(
    @field:NotBlank(message = "recipeName")
    val recipeName: String?,
    @field:NotEmpty(message = "ingredients") @field:Valid
    val ingredients: MutableSet<IngredientRequest>?
)
